modules/publicize/publicize.php

```diff
@@ -510,6 +510,210 @@ function test_publicize_conns() {
 	 */
 	abstract function test_connection( $service_name, $connection );
 
+	/**
+	 * Retrieves current list of connections and applies filters.
+	 *
+	 * Retrieves current available connections and checks if the connections
+	 * have already been used to share current post. Finally, the checkbox
+	 * form UI fields are calculated. This function exposes connection form
+	 * data directly as array so it can be retrieved for static HTML generation
+	 * or JSON consumption.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param integer $selected_post_id Optional. Post ID to query connection status for.
+	 *
+	 * @return array {
+	 *     Array of UI setup data for connection list form.
+	 *
+	 *     @type string 'unique_id'     ID string representing connection
+	 *     @type string 'service_name'  Slug of the connection's service (facebook, twitter, ...)
+	 *     @type string 'service_label' Service Label (Facebook, Twitter, ...)
+	 *     @type string 'display_name'  Connection's human-readable Username: "@jetpack"
+	 *     @type bool   'enabled'       Default value for the connection (e.g., for a checkbox).
+	 *     @type bool   'done'          Has this connection already been publicized to?
+	 *     @type bool   'toggleable'    Is the user allowed to change the value for the connection?
+	 *     @type bool   'global'        Is this connection a global one?
+	 * }
+	 */
+	public function get_filtered_connection_data( $selected_post_id = null ) {
+		$connection_list = array();
+
+		$post = get_post( $selected_post_id ); // Defaults to current post if $post_id is null.
+		// Handle case where there is no current post.
+		if ( ! empty( $post ) ) {
+			$post_id = $post->ID;
+		} else {
+			$post_id = null;
+		}
+
+		$services = $this->get_services( 'connected' );
+		$all_done = $this->done_sharing_post( $post_id );
+
+		// We don't allow Publicizing to the same external id twice, to prevent spam.
+		$service_id_done = (array) get_post_meta( $post_id, $this->POST_SERVICE_DONE, true );
+
+		foreach ( $services as $service_name => $connections ) {
+			foreach ( $connections as $connection ) {
+				$connection_meta = $this->get_connection_meta( $connection );
+				$connection_data = $connection_meta['connection_data'];
+
+				$unique_id = $this->get_connection_unique_id( $connection );
+
+
+				// Was this connection (OR, old-format service) already Publicized to?
+				$done = ! empty( $post ) && (
```

Comment:
Can we drop this check, and use `$post_id` instead of `$post->ID` below ([L569](https://github.com/Automattic/jetpack/pull/10396/files#diff-2f5f58835a08193f26adccabf4ae5ef3R567), [R569](https://github.com/Automattic/jetpack/pull/10396/files#diff-2f5f58835a08193f26adccabf4ae5ef3R569)) ? (Because of the check on [R544](https://github.com/Automattic/jetpack/pull/10396/files#diff-2f5f58835a08193f26adccabf4ae5ef3R544) and below) Or will `get_post_meta()` choke on a `null` post ID?